lib/rust/flexer/src/group.rs

```diff
@@ -111,120 +225,131 @@ impl From<&Group> for NFA {
 pub mod tests {
     extern crate test;
 
-    use crate::automata::dfa::DFA;
     use crate::automata::nfa;
-    use crate::automata::nfa::NFA;
     use crate::automata::pattern::Pattern;
-    use crate::group::Group;
+    use crate::group::{Group, GroupRegistry};
     use crate::group::rule::Rule;
 
     use std::default::Default;
     use test::Bencher;
+    use enso_prelude::default;
 
-    fn newline() -> Group {
+    fn newline() -> GroupRegistry {
         let     pattern = Pattern::char('\n');
         let mut group   = Group::default();
 
         group.add_rule(Rule{pattern,callback:"".into()});
 
-        group
+        let mut registry = GroupRegistry::default();
+        registry.add_group(group);
+        registry
     }
 
-    fn letter() -> Group {
+    fn letter() -> GroupRegistry {
         let     pattern = Pattern::range('a'..='z');
         let mut group   = Group::default();
 
         group.add_rule(Rule{pattern,callback:"".into()});
 
-        group
+        let mut registry = GroupRegistry::default();
+        registry.add_group(group);
+        registry
     }
 
-    fn spaces() -> Group {
+    fn spaces() -> GroupRegistry {
         let     pattern = Pattern::char(' ').many1();
         let mut group   = Group::default();
 
         group.add_rule(Rule{pattern,callback:"".into()});
 
-        group
+        let mut registry = GroupRegistry::default();
+        registry.add_group(group);
+        registry
```

Comment:
Using mutable variabnle just to add something to it and then return seems like a often needed thing. Thus it should be rewritten to `GroupRegistry::from(group)`, or even simpler (because you have output type defined in signature), just to `group.into()`.